bundles/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleCloudAPI.java

```diff
@@ -0,0 +1,469 @@
+     * @return the filename without the extension
+     */
+    private String removeExtension(String fileName) {
+        int extensionPos = fileName.lastIndexOf(EXTENSION_SEPARATOR);
+        int lastSeparator = Math.max(fileName.lastIndexOf(UNIX_SEPARATOR), fileName.lastIndexOf(WINDOWS_SEPARATOR));
+        return lastSeparator > extensionPos ? fileName : fileName.substring(0, extensionPos);
+    }
+
+    /**
+     * Call Google service to synthesize the required text
+     *
+     * @param text Text to synthesize
+     * @param voice Voice parameter
+     * @param audioFormat Audio encoding format
+     * @return Audio input stream or {@code null} when encoding exceptions occur
+     */
+    @SuppressWarnings({ "null", "unused" })
+    private byte[] synthesizeSpeechByGoogle(String text, GoogleTTSVoice voice, String audioFormat)
+            throws AuthenticationException, IOException {
+        AudioConfig audioConfig = new AudioConfig(AudioEncoding.valueOf(audioFormat), config.pitch, config.speakingRate,
+                config.volumeGainDb);
+        SynthesisInput synthesisInput = new SynthesisInput(text);
+        VoiceSelectionParams voiceSelectionParams = new VoiceSelectionParams(voice.getLocale().getLanguage(),
+                voice.getLabel(), SsmlVoiceGender.valueOf(voice.getSsmlGender()));
+
+        SynthesizeSpeechRequest request = new SynthesizeSpeechRequest(audioConfig, synthesisInput,
+                voiceSelectionParams);
+
+        HttpRequestBuilder builder = HttpRequestBuilder.postTo(SYTNHESIZE_SPEECH_URL)
+                .withHeader(HttpHeader.AUTHORIZATION.name(), getAuthorizationHeader())
+                .withContent(gson.toJson(request), MimeTypes.Type.APPLICATION_JSON.name());
+
+        SynthesizeSpeechResponse synthesizeSpeechResponse = gson.fromJson(builder.getContentAsString(),
+                SynthesizeSpeechResponse.class);
+
+        if (synthesizeSpeechResponse == null) {
+            return null;
+        }
+
+        byte[] encodedBytes = synthesizeSpeechResponse.getAudioContent().getBytes(StandardCharsets.UTF_8);
+        return Base64.getDecoder().decode(encodedBytes);
+    }
+
+    /**
+     * Gets a unique filename for a give text, by creating a MD5 hash of it. It
+     * will be preceded by the locale.
+     * <p>
+     * Sample: "en-US_00a2653ac5f77063bc4ea2fee87318d3"
+     */
+    private String getUniqueFilenameForText(String text, String voiceName) {
+        try {
+            MessageDigest md = MessageDigest.getInstance("MD5");
+            byte[] bytesOfMessage = (config.toConfigString() + text).getBytes(StandardCharsets.UTF_8);
+            byte[] md5Hash = md.digest(bytesOfMessage);
+            BigInteger bigInt = new BigInteger(1, md5Hash);
+            String fileNameHash = bigInt.toString(16);
+            // Now we need to zero pad it if you actually want the full 32 chars
+            while (fileNameHash.length() < 32) {
+                fileNameHash = "0" + fileNameHash;
```

Comment:
I think you can simplify the conversion to hexadecimal and left padding the 0's using

```java
String fileNameHash = String.format("%032x", bigInt);
```